src/main/java/com/dtforce/spring/kubernetes/KubernetesDiscoveryClient.java

```diff
@@ -49,33 +44,80 @@ public String description()
 			return Collections.emptyList();
 		}
 
-		if (log.isDebugEnabled()) {
-			log.debug("getInstances: service = {}", service.toString());
+		return getInstancesFromService(service);
+	}
+
+	@Override
+	public List<ServiceInstance> selectInstances(Map<String, String> match)
+	{
+		return selectInstances(match, Collections.emptyMap());
+	}
+
+	@Override
+	public List<ServiceInstance> selectInstances(Map<String, String> match, Map<String, String> doNotMatch)
+	{
+		ServiceList serviceList;
+		try {
+			serviceList = kubeClient.services().withLabels(match).withoutLabels(doNotMatch).list();
+		} catch(KubernetesClientException e) {
+			log.warn("getInstancesMatchingMetadata: failed to retrieve matching services: API call failed: {}",
+				e.getMessage());
+			return Collections.emptyList();
```

Comment:
It's question whether to swallow the exception here - what was your reasoning?

Reply:
I have to admit: swallowing the exception is not a good idea, as `KubernetesClientException` being thrown means that something is wrong with the API client or permissions of the underlying Service Account.